### ISSUE TITLE
utils/indicators.py

### ISSUE THREAD

**Issue by temur3559** [temur3559/UltimateCryptoBotV2#20]
import pandas as pd

def calculate_rsi(data, period=14):
    delta = data['close'].diff()
    gain = (delta.where(delta > 0, 0)).rolling(window=period).mean()
    loss = (-delta.where(delta < 0, 0)).rolling(window=period).mean()

    rs = gain / loss
    rsi = 100 - (100 / (1 + rs))
    return rsi

def calculate_ema(data, period=21):
    return data['close'].ewm(span=period, adjust=False).mean()

def calculate_macd(data):
    ema12 = calculate_ema(data, 12)
    ema26 = calculate_ema(data, 26)
    macd = ema12 - ema26
    signal = macd.ewm(span=9, adjust=False).mean()
    return macd, signal

def add_indicators(df):
    df['RSI'] = calculate_rsi(df)
    df['EMA21'] = calculate_ema(df, 21)
    df['EMA50'] = calculate_ema(df, 50)
    df['MACD'], df['MACD_Signal'] = calculate_macd(df)
    return df
